src/main/java/com/laytonsmith/core/functions/Commands.java

```diff
@@ -331,4 +335,120 @@ public Version since() {
 			return CHVersion.V3_3_1;
 		}
 	}
+
+	@api
+	public static class get_commands extends AbstractFunction {
+
+		@Override
+		public ExceptionType[] thrown() {
+			return new ExceptionType[0];
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return false;
+		}
+
+		@Override
+		public Construct exec(Target t, Environment environment, Construct... args) throws ConfigRuntimeException {
+			MCCommandMap map = Static.getServer().getCommandMap();
+			Collection<MCCommand> commands = map.getCommands();
+			CArray ret = CArray.GetAssociativeArray(t);
+			for(MCCommand command : commands) {
+				CArray ca = CArray.GetAssociativeArray(t);
+				ca.set("name", new CString(command.getName(), t), t);
+				ca.set("description", new CString(command.getDescription(), t), t);
+				Construct permission;
+				if (command.getPermission() == null) {
+					permission = CNull.NULL;
+				} else {
+					permission = new CString(command.getPermission(), t);
+				}
+				ca.set("permission", permission, t);
+				ca.set("nopermmsg", new CString(command.getPermissionMessage(), t), t);
+				ca.set("usage", new CString(command.getUsage(), t), t);
+				CArray aliases = new CArray(t);
+				for (String a : command.getAliases()) {
+					aliases.push(new CString(a, t));
+				}
+				ca.set("aliases", aliases, t);
+				ret.set(command.getName(), ca, t);
+			}
+			return ret;
+		}
+
+		@Override
+		public String getName() {
+			return "get_commands";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{0};
+		}
+
+		@Override
+		public String docs() {
+			return "array {} Returns an array of command arrays in the format register_command expects."
+					+ " This does not include CommandHelper aliases, as they are not registered commands.";
```

Comment:
Use Implementation.GetServerType().getBranding() here instead of hardcoding "CommandHelper".